tests/pages/base_page.py

```
@@ -64,19 +84,18 @@ def click(self, locator, timeout=None, retries=3):
 
-            except (StaleElementReferenceException, TimeoutException) as err:
-                last_exception = err
-
-        raise last_exception or TimeoutException(
-            f"Failed to click element after {retries} retries: {locator}"
-        )
+            except (StaleElementReferenceException, TimeoutException):
+                if attempt == retries - 1:
+                    raise TimeoutException(
+                        f"Failed to click {locator} after {retries} retries"
+                    )
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)